controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -52,11 +84,439 @@ func (r *StorageConsumerReconciler) Reconcile(ctx context.Context, request recon
+		return err
+	}
+
+	r.storageConsumer.Status.CephObjects.CephRBDProvisionerUser = r.cephClientRBDProvisioner.Name
+	return nil
+}
+
+func (r *StorageConsumerReconciler) reconcileCephClientRBDNode() error {
+
+	// TODO: Update caps relavent to rbd node
+	desired := &rookCephv1.CephClient{
+		Spec: rookCephv1.ClientSpec{
+			Caps: map[string]string{
+				"mon": "profile rbd",
+				"osd": fmt.Sprintf("profile rbd pool=%s", r.cephBlockPool.Name),
+			},
+		},
+	}
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.cephClientRBDNode, func() error {
+		if err := r.own(r.cephClientRBDNode); err != nil {
+			return err
+		}
+
+		r.cephClientRBDNode.Spec = desired.Spec
+		return nil
+	})
+
+	if err != nil {
+		r.Log.Error(err, "Failed to update CephClient.", "CephClient", klog.KRef(r.cephClientRBDNode.Namespace, r.cephClientRBDNode.Name))
+		return err
+	}
+
+	r.storageConsumer.Status.CephObjects.CephRBDNodeUser = r.cephClientRBDNode.Name
+	return nil
+}
+
+func (r *StorageConsumerReconciler) reconcileCephClientCephFSProvisioner() error {
+
+	// TODO: Update caps relavent to cephfs provisioner
+	desired := &rookCephv1.CephClient{
+		Spec: rookCephv1.ClientSpec{
+			Caps: map[string]string{
+				"mon": "profile rbd",
+				"osd": fmt.Sprintf("profile rbd pool=%s", r.cephBlockPool.Name),
+			},
+		},
+	}
+
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.cephClientCephFSProvisioner, func() error {
+		if err := r.own(r.cephClientCephFSProvisioner); err != nil {
+			return err
+		}
+
+		r.cephClientCephFSProvisioner.Spec = desired.Spec
+		return nil
+	})
+
+	if err != nil {
+		r.Log.Error(err, "Failed to update CephClient.", "CephClient", klog.KRef(r.cephClientCephFSProvisioner.Namespace, r.cephClientCephFSProvisioner.Name))
+		return err
+	}
+
+	r.storageConsumer.Status.CephObjects.CephFSProvisionerUser = r.cephClientCephFSProvisioner.Name
+	return nil
+}
+
+func (r *StorageConsumerReconciler) reconcileCephClientCephFSNode() error {
+
+	// TODO: Update caps relavent to cephfs node
+	desired := &rookCephv1.CephClient{
+		Spec: rookCephv1.ClientSpec{
+			Caps: map[string]string{
+				"mon": "profile rbd",
+				"osd": fmt.Sprintf("profile rbd pool=%s", r.cephBlockPool.Name),
+			},
+		},
+	}
+
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.cephClientCephFSNode, func() error {
+		if err := r.own(r.cephClientCephFSNode); err != nil {
+			return err
+		}
+
+		r.cephClientCephFSNode.Spec = desired.Spec
+		return nil
+	})
+
+	if err != nil {
+		r.Log.Error(err, "Failed to update CephClient.", "CephClient", klog.KRef(r.cephClientCephFSNode.Namespace, r.cephClientCephFSNode.Name))
+		return err
+	}
+
+	r.storageConsumer.Status.CephObjects.CephFSNodeUser = r.cephClientCephFSNode.Name
+	return nil
+}
+
+func (r *StorageConsumerReconciler) reconcileCephClientHealthChecker() error {
+
+	// TODO: Update caps relavent to health checker
+	desired := &rookCephv1.CephClient{
+		Spec: rookCephv1.ClientSpec{
+			Caps: map[string]string{
```

Comment:
Rook uses:

```
            'mgr': 'allow command config',
            'mon': 'allow r, allow command quorum_status, allow command version',
            'osd': "allow rwx pool={0}.rgw.meta, " +
            "allow r pool=.rgw.root, " +
            "allow rw pool={0}.rgw.control, " +
            "allow rx pool={0}.rgw.log, " +
            "allow x pool={0}.rgw.buckets.index"
```

Forget about the pool rgw caps if there is no object store and stick with the first 2 lines.

Reply:
yes there is no object store , fixed